File: analysis/import/SCMLogParser/src/main/java/de/maibornwolff/codecharta/model/input/metrics/NumberOfOccurencesInCommits.java
package de.maibornwolff.codecharta.model.input.metrics;

import de.maibornwolff.codecharta.model.input.Modification;

public final class NumberOfOccurencesInCommits implements ModificationMetric<Integer> {
    private int numberOfOccurrencesInCommits = 0;

    @Override
    public String metricName() {
        return "number_of_commits";
    }

    @Override
    public void registerModification(Modification modification) {
        numberOfOccurrencesInCommits++;
    }

    @Override
    public Integer value() {
        return numberOfOccurrencesInCommits;
    }
}

File: analysis/import/SCMLogParserV2/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparserv2/input/metrics/NumberOfOccurencesInCommits.kt
package de.maibornwolff.codecharta.importer.scmlogparserv2.input.metrics

import de.maibornwolff.codecharta.importer.scmlogparserv2.input.Modification

class NumberOfOccurencesInCommits : Metric {
    private var numberOfOccurrencesInCommits: Long = 0

    override fun description(): String {
        return "Number Of Commits: Number of times this file occured in a commit."
    }

    override fun metricName(): String {
        return "number_of_commits"
    }

    override fun registerModification(modification: Modification) {
        numberOfOccurrencesInCommits++
    }

    override fun value(): Number {
        return numberOfOccurrencesInCommits
    }
}
